### ISSUE TITLE
wp-now: Trying to run a blueprint with wp-cli step fails

### ISSUE THREAD

**Issue by n3f** [WordPress/playground-tools#267]
## TL;DR

I'm looking at adding the ability to test/develop a plugin using `wp-now` but when I try to include a blueprint with some configuration steps it fails with `Error: Could not write to "/wordpress/run-cli.php": There is no such file or directory OR the parent directory does not exist.`.

## Details

From my plugin directory I added [this blueprint](https://github.com/WordPress/blueprints/blob/trunk/blueprints/posts-via-wp-cli/blueprint.json) (it's an example one).

I then ran `npx @wp-now/wp-now start  --blueprint=./blueprint.json --reset` (also ran without `--reset`).

That command then fails with the following output:

```txt
$ npx @wp-now/wp-now start  --blueprint=./blueprint.json --reset
Starting the server......
directory: /Users/brent/tmp/plugin-dir
mode: plugin
php: 8.0
wp: latest
WordPress latest folder already exists. Skipping download.
SQLite folder already exists. Skipping download.
Created a fresh SQLite database and wp-content directory.
blueprint steps: 2
Blueprint step completed: login
Blueprint step completed: writeFile
Error: Could not write to "/wordpress/run-cli.php": There is no such file or directory OR the parent directory does not exist.
    at descriptor.value (/Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@php-wasm/node/index.cjs:72672:17)
    at Object.wpCLI$1 [as wp-cli] (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:1209:87)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async o (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:9413:30)
    at async Object.run (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:9340:23)
    at async runBlueprintSteps (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:9448:3)
    at async startWPNow (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-now/wp-now/main.js:663:5)
    at async startServer (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-now/wp-now/main.js:981:42)
    at async Object.handler (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-now/wp-now/main.js:1167:25) {
  [cause]: [<generic error, no stack>] {
    name: 'ErrnoError',
    node: undefined,
    setErrno: [Function (anonymous)],
    errno: 44,
    message: 'FS error'
  }
}
Error: Error when executing the blueprint step #2 ({"step":"wp-cli","command":"wp post generate --count=12 --post_type=post --post_date=1999-01-04"}) : Could not write to "/wordpress/run-cli.php": There is no such file or directory OR the parent directory does not exist.
    at Object.run (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:9343:37)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runBlueprintSteps (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:9448:3)
    ... 2 lines matching cause stack trace ...
    at async Object.handler (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-now/wp-now/main.js:1167:25) {
  [cause]: Error: Could not write to "/wordpress/run-cli.php": There is no such file or directory OR the parent directory does not exist.
      at descriptor.value (/Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@php-wasm/node/index.cjs:72672:17)
      at Object.wpCLI$1 [as wp-cli] (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:1209:87)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async o (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:9413:30)
      at async Object.run (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:9340:23)
      at async runBlueprintSteps (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-playground/blueprints/index.js:9448:3)
      at async startWPNow (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-now/wp-now/main.js:663:5)
      at async startServer (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-now/wp-now/main.js:981:42)
      at async Object.handler (file:///Users/brent/.npm/_npx/7c65af5c66ec2a7e/node_modules/@wp-now/wp-now/main.js:1167:25) {
    [cause]: [<generic error, no stack>] {
      name: 'ErrnoError',
      node: undefined,
      setErrno: [Function (anonymous)],
      errno: 44,
      message: 'FS error'
    }
  }
}
Failed to start the server: Error when executing the blueprint step #2 ({"step":"wp-cli","command":"wp post generate --count=12 --post_type=post --post_date=1999-01-04"}) : Could not write to "/wordpress/run-cli.php": There is no such file or directory OR the parent directory does not exist.
```

**Comment by sejas** [comment]
Hi @n3f !, thanks for reporting. `wp-cli` is not integrated yet in `wp-now`, but is something we are planning to support in the next weeks. Once is integrated we can add support for the blueprint `"step": "wp-cli"`.

**Comment by n3f** [comment]
@sejas Any development towards supporting this?

**Comment by sejas** [comment]
Hi @n3f, we haven’t added support for the wp-cli blueprint step to wp-now.

[There’s a suggestion about deprecating wp-now](https://github.com/WordPress/wordpress-playground/issues/2100#issue-2747747224) in favor of [@wp-playground/cli](https://github.com/WordPress/wordpress-playground/tree/trunk/packages/playground/cli), but it also fails in the same step:

```
❯ npx @wp-playground/cli server --blueprint=./b.json

Starting a PHP server...
Setting up WordPress latest
Resolved WordPress release URL: https://downloads.w.org/release/wordpress-6.7.1.zip
Booting WordPress...
Booted!
Running the Blueprint...
Downloading wp-cli.phar – 100%
Error: PHP Deprecated:  WP_CLI\Runner::get_subcommand_suggestion(): Implicitly marking parameter $root_command as nullable is deprecated, the explicit nullable type must be used instead in phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php on line 1958

    at Object.Ar [as wp-cli] (file:///Users/macbookpro/.npm/_npx/8ab79f7a2a3f35a4/node_modules/@wp-playground/blueprints/index.js:643:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async d (file:///Users/macbookpro/.npm/_npx/8ab79f7a2a3f35a4/node_modules/@wp-playground/blueprints/index.js:12472:30)
    at async Object.run (file:///Users/macbookpro/.npm/_npx/8ab79f7a2a3f35a4/node_modules/@wp-playground/blueprints/index.js:12403:23)
    at async ns (file:///Users/macbookpro/.npm/_npx/8ab79f7a2a3f35a4/node_modules/@wp-playground/blueprints/index.js:12507:3)
    at async Object.onBind (file:///Users/macbookpro/.npm/_npx/8ab79f7a2a3f35a4/node_modules/@wp-playground/cli/cli.js:20:568)
    at async T (file:///Users/macbookpro/.npm/_npx/8ab79f7a2a3f35a4/node_modules/@wp-playground/cli/cli.js:1:1159)
file:///Users/macbookpro/.npm/_npx/8ab79f7a2a3f35a4/node_modules/@wp-playground/blueprints/index.js:12406
            throw fe.error(b), new Error(
                               ^

Error: Error when executing the blueprint step #2 ({"step":"wp-cli","command":"wp post generate --count=12 --post_type=post --post_date=1999-01-04"}) : PHP Deprecated:  WP_CLI\Runner::get_subcommand_suggestion(): Implicitly marking parameter $root_command as nullable is deprecated, the explicit nullable type must be used instead in phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php on line 1958
```

Given the current situation, I'm not sure if any of those tools are receiving maintenance.

What’s your use case? The [blueprint](https://github.com/WordPress/blueprints/blob/trunk/blueprints/posts-via-wp-cli/blueprint.json) you’re using runs a wp-cli command to create multiple posts:
`wp post generate --count=12 --post_type=post --post_date=1999-01-04`.

Maybe Studio could work for you. You can ask the Studio Assistant to run that command, or create a site in Studio or Playground Web, create a backup, and import it into a new Studio site as needed.

**Comment by n3f** [comment]
Sorry it took me so long to come back to this.

I was really just looking at trying to launch an e2e testing environment that I could populate with fixtures easily. I was able to do the wp-cli step without any problems on the playground site so I just assumed it would be easy to work here.  I can most likely implement my use case with `wp-env` but I liked the one-stop shop aspect of playground/wp-now.

**Comment by sejas** [comment]
I added the idea of using Studio CLI for e2e tests pdKhl6-4zz-p2#comment-7781. The best option for now would be to ask the Playground team if [@wp-playground/cli](https://github.com/WordPress/wordpress-playground/tree/trunk/packages/playground/cli) supports or plans to support this use case, although core contributions have slowed down for the time being.

**Comment by bgrgicak** [comment]
I'm working on a _Building Automated Tests with WordPress Playground_ workshop and plan to add the necessary support for e2e and integration tests to the Playground CLI. I expect to have it working in June.
